kasa/smartdimmer.py

```diff
@@ -41,19 +41,37 @@ def brightness(self) -> int:
         return int(sys_info["brightness"])
 
     @requires_update
-    async def set_brightness(self, value: int):
-        """Set the new dimmer brightness level in percentage."""
+    async def set_brightness(self, brightness: int, transition: int = None):
+        """Set the new dimmer brightness level in percentage.
+
+        :param int transition: duration in milliseconds to transition to the new brightness (default: None)
```

Comment:
I suppose `None` means in this case that the default transition is used? So passing `0` here would disable the transition altogether?

Reply:
Passing `0` will raise a ValueError, as the switch does not support a transition of 0.  

One subtlety between calling set_brightness() with a transition and without is that with the transition the switch automatically turns on, but without the transition the switch does not automatically turn on.

Reply:
Hmm, if the dimmers are any similar to the bulbs, i.e., there is a short transition when turning the bulb on/off, to get an immediate change (without the regular transition when nothing is defined), one should pass a very short transition?

Please add a note to the docstring that using transition will turn the device on. Having the ability to adjust the settings while the device is turned off is expected and makes sense in many cases.